cypress/component/PncDetails.cy.tsx

```diff
@@ -0,0 +1,172 @@
+import { CourtCaseContext } from "context/CourtCaseContext"
+import { CurrentUserContext } from "context/CurrentUserContext"
+import { DisplayFullCourtCase } from "types/display/CourtCases"
+import { DisplayFullUser } from "types/display/Users"
+import Permission from "types/Permission"
+import PncDetails from "../../src/features/CourtCaseDetails/Sidebar/Pnc/PncDetails"
+
+describe("PNC details", () => {
+  const currentUser = {
+    username: "",
+    email: "",
+    visibleForces: [],
+    visibleCourts: [],
+    excludedTriggers: [],
+    featureFlags: {},
+    groups: [],
+    hasAccessTo: {
+      [Permission.CaseDetailsSidebar]: false,
+      [Permission.Exceptions]: false,
+      [Permission.Triggers]: false,
+      [Permission.UnlockOtherUsersCases]: false,
+      [Permission.ListAllCases]: false,
+      [Permission.ViewReports]: false,
+      [Permission.ViewUserManagement]: false
+    }
+  } as DisplayFullUser
+
+  it("displays all PNC data", () => {
+    const pncQueryData = {
+      forceStationCode: "01ZD",
+      checkName: "LEBOWSKI",
+      pncId: "2021/0000006A",
+      courtCases: [
+        {
+          courtCaseReference: "21/2732/000006N",
+          offences: [
+            {
+              offence: {
+                acpoOffenceCode: "5:5:5:1",
+                cjsOffenceCode: "TH68001",
+                title: "Theft from the person of another",
+                sequenceNumber: 1,
+                qualifier1: "Q1",
+                qualifier2: "Q2",
+                startDate: "2010-11-28T00:00:00.000Z",
+                endDate: "2010-12-31T00:00:00.000Z"
+              },
+              adjudication: {
+                verdict: "GUILTY",
+                sentenceDate: "2011-09-25T00:00:00.000Z",
+                plea: "NOT GUILTY",
+                offenceTICNumber: 321
+              },
+              disposals: [
+                {
+                  qtyDate: "2012-01-25T00:00:00.000Z",
+                  qtyDuration: "Y999",
+                  type: 2007,
+                  qtyUnitsFined: "",
+                  qtyMonetaryValue: "1000",
+                  qualifiers: "AA",
+                  text: "This is a dummy text"
+                }
+              ]
+            }
+          ],
+          crimeOffenceReference: "XOXO"
+        }
+      ]
+    }
+
+    const courtCase = {
+      aho: {
+        PncQuery: pncQueryData,
+        PncQueryDate: "2024-07-10T00:00:00.000Z"
+      }
+    } as unknown as DisplayFullCourtCase
+
+    cy.mount(
+      <CourtCaseContext.Provider value={[{ courtCase, amendments: {}, savedAmendments: {} }, () => {}]}>
+        <CurrentUserContext.Provider value={{ currentUser }}>
```

Comment:
Small thing: In the `court-cases/[courtCaseId]/index.tsx` page the provider is the other way round. E.g.

```jsx
<CurrentUserContext.Provider value={currentUserContext}>
  <CourtCaseContext.Provider value={courtCaseContext}>
  </CourtCaseContext.Provider>
</CurrentUserContext.Provider>
```